src/irmin/contents.mli

```diff
@@ -16,8 +16,54 @@
 
 (** Values. *)
 
+type json = [
+  | `Null
+  | `Bool of bool
+  | `String of string
+  | `Float of float
+  | `O of (string * json) list
+  | `A of json list
+]
+
 module String: S.CONTENTS with type t = string
 module Cstruct: S.CONTENTS with type t = Cstruct.t
+module Json: S.CONTENTS with type t = (string * json) list
+module Json_value: S.CONTENTS with type t = json
+module Json_tree(P: S.PATH)(M: S.METADATA): sig
+  include S.CONTENTS with type t = json
+  module type STORE = S.STORE with
+    type contents = t
+    and type key = P.t
+    and type step = P.step
+    and type metadata = M.t
+
+  type tree = [
+    | `Tree of (P.step * tree) list
+    | `Contents of json * M.t
```

Comment:
The `M.t` part here doesn't seem to bring much. I think I would simply fix `STORE.metadata = ()` or something like that.